server/controllers/user.js

```diff
@@ -2,68 +2,53 @@ const User = require("../models/User");
 const asyncHandler = require("express-async-handler");
 
 // @route POST /users
-// @desc Search for users
-// @access Private
-exports.searchUsers = asyncHandler(async (req, res, next) => {
-  const searchString = req.query.search;
+// @desc Checks if user exists
+// @access Public
+exports.doesUserExist = asyncHandler(async (req, res, next) => {
+  const { email } = req.body;
 
-  let users;
-  if (searchString) {
-    users = await User.find({
-      username: { $regex: searchString, $options: "i" }
-    });
-  }
+  const user = await User.findOne({ email: email });
 
-  if (!users) {
-    res.status(404);
-    throw new Error("No users found in search");
+  if (!user) {
+    res.status(401);
+    throw new Error(`No account exists with the email: ${email}`);
   }
 
-  res.status(200).json({ users: users });
+  res.status(200).json({
+    success: {
+      message: "Account exists"
+    }
+  });
 });
 
-// @route GET /users/:username
-// @desc Get a user by username
-// @access Private
+// @route GET /users/me
+// @desc Sends the logged in user's information back
+// @access Public
 exports.getUser = asyncHandler(async (req, res, next) => {
-    const username = req.params.username
-    const user = await User.findOne({ username });
-
-    if (!user) {
-        res.status(404);
-        throw new Error("User not found" );
-    }
-
-    res.status(200).json({
-        success: {
-          user: {
-            id: user._id,
-            username: user.username,
-            email: user.email,
-            register_date: user.register_date
-       }
-    }
-  });
+  res.send(req.user);
 });
 
 // @route PATCH /users/:id
 // @desc Update user
 // @access Private
 exports.updateUser = asyncHandler(async (req, res, next) => {
-    const id = req.params.id;
-    updates = req.body;
-    options = { new: true };
-
-    const user = await User.findByIdAndUpdate(id, updates, options);
-
-    res.status(200).json({
-        success: {
-          user: {
-            id: user._id,
-            username: user.username,
-            email: user.email,
-            register_date: user.register_date
-       }
+  const id = req.params.id;
+  updates = req.body;
+  options = { new: true };
+
+  const user = await User.findByIdAndUpdate(id, updates, options);
```

Comment:
This works, but just curious on the integrity of our database without sanitization or validation on body object? Could we refactor this in the future to do more safety on the data being provided?

Reply:
For example, if a user updates their email then we should make sure they are sending down an actual email in the request body.

Reply:
Just refactored it! I updated the user model and created a middleware to check for valid email/usernames. The model now only accepts email/usernames that is in lowercase.